src/json.ts

```diff
@@ -266,3 +272,166 @@ export function getJsonRpcIdValidator(options?: JsonRpcValidatorOptions) {
   };
   return isValidJsonRpcId;
 }
+
+/**
+ * Checks whether a value is JSON serializable and counts the total number
+ * of bytes needed to store the serialized version of the value.
+ *
+ * @param jsObject - Potential JSON serializable object.
+ * @param skipSizingProcess - Skip JSON size calculation (default: false).
+ * @returns Tuple [isValid, plainTextSizeInBytes] containing a boolean that signals whether
+ * the value was serializable and a number of bytes that it will use when serialized to JSON.
+ */
+export function validateJsonAndGetSize(
+  jsObject: unknown,
+  skipSizingProcess = false,
+): [isValid: boolean, plainTextSizeInBytes: number] {
+  const seenObjects: unknown[] = [];
+  /**
+   * Checks whether a value is JSON serializable and counts the total number
+   * of bytes needed to store the serialized version of the value.
+   *
+   * This function assumes the encoding of the JSON is done in UTF-8.
+   *
+   * @param value - Potential JSON serializable value.
+   * @param skipSizing - Skip JSON size calculation (default: false).
+   * @returns Tuple [isValid, plainTextSizeInBytes] containing a boolean that signals whether
+   * the value was serializable and a number of bytes that it will use when serialized to JSON.
+   */
+  function getJsonSerializableInfo(
+    value: unknown,
+    skipSizing: boolean,
+  ): [isValid: boolean, plainTextSizeInBytes: number] {
+    if (value === undefined) {
+      // Return zero for undefined, since these are omitted from JSON serialization
+      return [true, 0];
+    } else if (value === null) {
+      // Return already specified constant size for null (special object)
+      return [true, skipSizing ? 0 : JsonSize.Null];
+    }
+
+    // Check and calculate sizes for basic types
+    // eslint-disable-next-line default-case
+    switch (typeof value) {
+      case 'function':
+        return [false, 0];
+      case 'string':
+        return [
+          true,
+          skipSizing ? 0 : calculateStringSize(value) + JsonSize.Quote * 2,
+        ];
+      case 'boolean':
+        if (skipSizing) {
+          return [true, 0];
+        }
+        return [true, value ? JsonSize.True : JsonSize.False];
+      case 'number':
+        return [true, skipSizing ? 0 : calculateNumberSize(value)];
+    }
+
+    // Handle specific complex objects that can be serialized properly
+    try {
+      if (value instanceof Date) {
+        if (skipSizing) {
+          return [true, 0];
+        }
+        const jsonSerializedDate = value.toJSON();
```

Comment:
I wonder if there is a better way to do this 🤔

Reply:
Actually, there is. Your question was a lead to a better solution. Updated now.